battle-engine.js

```diff
@@ -541,6 +541,18 @@ BattlePokemon = (function () {
 				target = this.battle.runEvent('RedirectTarget', this, this, move, target);
 			}
 			targets = [target];
+
+			// Resolve apparent targets for Pressure.
+			if (move.pressureTarget) {
```

Comment:
Because we don't want the "targets" that appear from `move.pressureTarget` to be first on the targets list. The `'self'` in `move.target` for Imprison/Snatch needs to come first so the system can account for everything Pressure-related properly while still officially targetting the user of the move.

It's a hack, it's all a hack. The only thing I can think of that would work otherwise is if we changed `move.target` to `'selfAndFoeSide'` and had a special handler for that?